packages/react/README.md

```diff
@@ -101,3 +119,593 @@ function WidgetDeleter() {
+- `options`: Options for making the call to the backend. Not required.
+  - `requestPolicy`: The `urql` request policy to make the request with. See [`urql`'s docs](https://formidable.com/open-source/urql/docs/api/urql/);
+  - `pause`: Should the hook make a request right now or not. See [`urql`'s docs](https://formidable.com/open-source/urql/docs/api/urql/)
+
+`useGlobalAction` returns two values: a result object with the `data`, `fetching`, and `error` keys for inspecting in your React component's output, and a `act` function to actually run the backend global action. `useGlobalAction` is a rule-following React hook that wraps action execution, which means it doesn't just run the action as soon as the hook is invoked. Instead, `useGlobalAction` returns a configured function which you need to call in response to some event. This `act` function accepts the action inputs as arguments. `useGlobalAction`'s result will return the `data`, `fetching`, and `error` details for the most recent execution of the action.
+
+- `data: Record<string, any> | null`: The data returned by the global action from the backend. Is `null` while before the mutation is run and while it is currently ongoing.
+- `fetching: boolean`: A boolean describing if the hook is currently making a request to the backend.
+- `error: Error | null`: An error from the client or server side, if encountered during the mutation. Will contain an error if the client passed invalid data, if server failed to complete the mutation, or if a network error was encountered. See the [Errors](#errors-from-the-returned-error-object) section.
+
+For example, we can create a button which runs a global action called `purgeData` when clicked
+
+```javascript
+import React from "react";
+import { api } from "../api"; // some file that instantiates @gadget-client/your-app-slug
+
+export const PurgeData = () => {
+  // invoke the `useGlobalAction` hook, getting back a result object and an action runner function every render
+  const [{ data, fetching, error }, runPurge] = useGlobalAction(api.purgeData);
+
+  // deal with all the states of the result object
+  if (fetching) return "Loading...";
+  if (error) return `Error running global action: ${error}`;
+
+  if (!data) {
+    return (
+      <button
+        onClick={() => {
+          // run the global action when the button is clicked
+          // the action runner function accepts action inputs in the same format as api.purgeData, and the GraphQL API
+          purgeData({ foo: "bar" });
+        }}
+      >
+        Purge data
+      </button>
+    );
+  } else {
+    return "Purge completed";
+  }
+};
+```
+
+### `useGet(singletonModelManager: SingletonModelManager, options: GetOptions = {}): [{data, fetching, error}, refetch]`
```

Comment:
Just a mental note I'm gonna poke away: if we never really have any other singletons, we should perhaps consider more useful hooks like `useCurrentSession` and `useCurrentUser`.